anchore_engine/services/policy_engine/engine/feeds/config.py

```diff
@@ -0,0 +1,284 @@
+              nvdv2:
+                enabled: true
+                url: https://ancho.re/v1/service/feeds
+              github:
+                enabled: true
+                url: https://ancho.re/v1/service/feeds
+
+    :return: dict that is the vulnerabilities section of config
+    """
+    full_config = localconfig.get_config()
+    vuln_config = (
+        full_config.get("services", {})
+        .get(
+            "policy_engine", {}
+        )  # TODO check if there's a better way to find service config
+        .get("vulnerabilities", {})
+    )
+
+    return vuln_config if vuln_config is not None else {}
+
+
+def get_selected_configs_to_sync(vulnerabilities_config: Dict) -> Dict[str, SyncConfig]:
+    """
+    Returns the feeds to be synced based on the configuration.
+    Handles both legacy and grype vulnerabilities provider configurations
+
+    :param vulnerabilities_config: vulnerabilities section of config file
+    {
+       "provider": "legacy",
+       "sync": {
+          "enabled": True,
+          "ssl_verify": True,
+          "connection_timeout_seconds": 3,
+          "read_timeout_seconds": 60,
+          "data": {
+             "vulnerabilities": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             },
+             "nvdv2": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             },
+             "github": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             }
+          }
+       }
+    }
+    :return: dictionary with feed names each mapped to its respective SyncConfig
+    """
+    # check for the provider first
+    if not vulnerabilities_config:
+        vulnerabilities_config = {}
+
+    sync_config = vulnerabilities_config.get("sync", {})
+    data_config = sync_config.get("data", {})
+    provider = vulnerabilities_config.get("provider")
+
+    if not provider or provider not in DEFAULT_PROVIDER_SYNC_CONFIG_MAP:
+        logger.warn(
+            "%s is not a supported provider. Defaulting to %s",
+            provider,
+            DEFAULT_VULNERABILITIES_PROVIDER,
+        )
+        provider = DEFAULT_VULNERABILITIES_PROVIDER
+
+    connection_timeout_seconds = sync_config.get("connection_timeout_seconds", 3)
+    read_timeout_seconds = sync_config.get("read_timeout_seconds", 60)
+
+    if data_config:
+        logger.debug(
+            "Searching for data feeds to be synced for %s provider in %s",
+            provider,
+            data_config,
+        )
+        to_be_synced = dict()
+        for feed_name, sync_config_dict in data_config.items():
+            # get the default config for both verifying feed_name is a valid data feed and accessing url if necessary
+            default_sync_config = DEFAULT_PROVIDER_SYNC_CONFIG_MAP.get(provider).get(
+                feed_name
+            )
+            if default_sync_config:
```

Comment:
because of the earlier block defaulting the provider, this is guaranteed to not be None. Also would reduce nesting if you checked the negative (! default_sync_config) and unindented the true case.

Reply:
`.get(feed_name)` can result in `None` if `feed_name` is not a supported feed for the provider. So default_sync_config is *not* guaranteed to be not None

Reply:
i'm referring to lines 134-140

Reply:
```
    if not provider or provider not in DEFAULT_PROVIDER_SYNC_CONFIG_MAP:
        logger.warn(
            "%s is not a supported provider. Defaulting to %s",
            provider,
            DEFAULT_VULNERABILITIES_PROVIDER,
        )
        provider = DEFAULT_VULNERABILITIES_PROVIDER
```